src/frontend/screens/CreateProject/index.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import { StyleSheet, View } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+
+import { WHITE } from "../../lib/styles";
+import Button from "../../sharedComponents/Button";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+import Text from "../../sharedComponents/Text";
+
+import { InputField } from "./InputField";
+import { useInputFieldValue } from "./useFieldValue";
+
+const m = defineMessages({
+  createProjectTitle: {
+    id: "screens.CreateProject.createProjectTitle",
+    defaultMessage: "Create a Project",
+    description: "Screen title for Create Project flow",
+  },
+  projectNameLabel: {
+    id: "screens.CreateProject.projectNameLabel",
+    defaultMessage: "Enter a name for the Project",
+    description: "Label text for project name input field",
+  },
+  deviceNameLabel: {
+    id: "screens.CreateProject.deviceNameLabel",
+    defaultMessage: "Add a name for your device",
+    description: "Label text for device name input field",
+  },
+  projectNamePlaceholder: {
+    id: "screens.CreateProject.projectNamePlaceholder",
+    defaultMessage: "Project Name",
+    description: "Placeholder text for project name input field",
+  },
+  deviceNamePlaceholder: {
+    id: "screens.CreateProject.deviceNamePlaceholder",
+    defaultMessage: "Device Name",
+    description: "Placeholder text for device name input field",
+  },
+  createProjectButton: {
+    id: "screens.CreateProject.createProjectButton",
+    defaultMessage: "Create Project",
+    description: "Text for submit button",
+  },
+  projectNameError: {
+    id: "screens.CreateProject.projectNameError",
+    defaultMessage: "Enter Project Name",
+    description: "Validation error for project name input field",
+  },
+  deviceNameError: {
+    id: "screens.CreateProject.deviceNameError",
+    defaultMessage: "Enter Device Name",
+    description: "Validation error for device name input field",
+  },
+});
+
+export const CreateProjectScreen: NavigationStackScreenComponent = ({
+  navigation,
+}) => {
+  const { formatMessage: t } = useIntl();
+
+  const [
+    projectName,
+    setProjectName,
+    projectNameHasError,
+    validateProjectName,
+  ] = useInputFieldValue();
+  const [
+    deviceName,
+    setDeviceName,
+    deviceNameHasError,
+    validateDeviceName,
+  ] = useInputFieldValue();
+
+  const updateProjectName = React.useCallback(value => {
+    setProjectName(value);
+  }, []);
+
+  const updateDeviceName = React.useCallback(value => {
+    setDeviceName(value);
+  }, []);
```

Comment:
Are these useCallbacks necessary?

Reply:
probably not. gotten into the nasty habit of defaulting to that but it's not necessary in this case. will fix 👍